src/main/webapp/app/modeling-editor/modeling-editor.component.ts

```diff
@@ -117,4 +125,60 @@ export class ModelingEditorComponent implements AfterViewInit, OnDestroy, OnChan
             this.apollonEditor.destroy();
         }
     }
+
+    /**
+     * Resets the UML model for the guided tour by removing the elements, relationships and assessments
+     * @param umlModel the model that should be reset
+     */
+    private resetUMLModelForGuidedTour(umlModel: UMLModel): void {
+        if (umlModel) {
+            umlModel.elements = [];
+            umlModel.relationships = [];
+            umlModel.assessments = [];
+        }
+        this.initializeApollonEditor();
+    }
+
+    /**
+     * Subscribes to the guided tour service
+     */
+    private subscribeForUMLModelReset() {
+        this.guidedTourService.resetUMLModel().subscribe(reset => {
+            if (reset) {
+                this.resetUMLModelForGuidedTour(this.umlModel);
+            }
+        });
+    }
+
+    /**
+     * Assess the model for the modeling guided tutorial
+     * @param umlName  the identifier of the UML element that has to be assessed
+     * @param umlModel  the current UML model in the editor
+     */
+    assessModelForGuidedTour(umlName: string, umlModel: UMLModel): void {
+        // Find the required UML classes
+        const personClass = umlModel.elements.find(element => element.name.trim() === personUML.name && element.type === 'Class');
+        const studentClass = umlModel.elements.find(element => element.name.trim() === studentUML.name && element.type === 'Class');
```

Comment:
I think in this case it is ok to not use `lodash` since we don't use complicated (calculation) logic in here that could be simplified by using lodash functions. Moreover I found this article here which would prefer using JavaScript instead of loadash for performance reasons: https://codeburst.io/why-you-shouldnt-use-lodash-anymore-and-use-pure-javascript-instead-c397df51a66

Reply:
Still I have to disagree here. The article you added is very misleading, it also tests the time used to import lodash, while we basically already pay this price once, since we are using it in other places. In real life usage lodash is just a tiny bit slower, but it checks for more corner cases, that could lead to exceptions. Here is a better article I found https://hackernoon.com/es6-lodash-native-javascript-nodejs-example-tutorial-map-function-functional-programming-performance-speed-69645e613049